fs/ensure_symlink.ts

```diff
@@ -4,15 +4,28 @@ import { ensureDir, ensureDirSync } from "./ensure_dir.ts";
 import { getFileInfoType, toPathString } from "./_util.ts";
 import { isWindows } from "../_util/os.ts";
 
+export function resolveSymlinkSrc(src: string, dest: string): string;
+export function resolveSymlinkSrc(src: URL, dest: URL): URL;
+export function resolveSymlinkSrc(src: string | URL, dest: string | URL) {
+  if (typeof src == "string" && typeof dest == "string") {
+    return path.resolve(path.dirname(dest), src);
+  } else {
+    return new URL(src, dest)
+  }
+}
+
 /**
  * Ensures that the link exists.
  * If the directory structure does not exist, it is created.
  *
  * @param src the source file path
  * @param dest the destination link path
  */
-export async function ensureSymlink(src: string | URL, dest: string | URL) {
-  const srcStatInfo = await Deno.lstat(src);
+export function ensureSymlink(src: string, dest: string);
+export function ensureSymlink(src: URL, dest: URL);
+export function ensureSymlink(src: any, dest: any) {
```

Comment:
I think we shouldn't change the typing here. This now doesn't support the usage like `ensureSymlink(src: string, dest: URL)` which we currently support.

I think we should align `src` and `dest` to either URL or string by checking each type

Reply:
I'm not sure why both URL and string are supported.

Does creating symlink to URL make any sense?